src/renderer/lib/dolphin.ts

```diff
@@ -0,0 +1,92 @@
+import * as path from "path";
+import {execFile, ChildProcess} from "child_process";
+import { remote } from "electron";
+
+import { dispatcher, store } from '@/store';
+import {setRecordingState, OBSRecordingAction} from '@/lib/obs'
+
+export interface DolphinState {
+    currentFrame: number,
+    lastGameFrame: number,
+    startRecordingFrame: number,
+    endRecordingFrame: number,
+    recordingStarted: boolean,
+};
+
+const dolphinState: DolphinState = {
+    currentFrame: -124,
+    lastGameFrame: -124,
+    startRecordingFrame: -124,
+    endRecordingFrame: -124,
+    recordingStarted: false,
+}
+
+const resetState = () => {
+    dolphinState.currentFrame = -124;
+    dolphinState.lastGameFrame = -124;
+    dolphinState.startRecordingFrame = -124;
+    dolphinState.endRecordingFrame = -124;
+}
+
+export const openComboInDolphin = (comboFilePath: string): void => {
+    const appData = remote.app.getPath("appData");
+    const dolphinPath = path.join(appData, "Slippi Desktop App", "dolphin", "Dolphin.exe");
+    console.log(dolphinPath);
+    store.getState().tempContainer.dolphin?.kill();
+    const dolphin: ChildProcess = execFile(dolphinPath, ["-i", comboFilePath], {maxBuffer: 2**20});
+    dispatcher.tempContainer.setDolphin(dolphin);
+    if (store.getState().tempContainer.obsConnected && store.getState().tempContainer.recordReplays) {
+        dolphin.stdout?.on('data', dolphinStdoutHandler);
+    }
+};
+
+const dolphinStdoutHandler = (line: string) => {
+    const commands: string[] = line.split("\r\n").filter((value: string) => value);
```

Comment:
Should the `\r\n` be OS agnostic using something like `os.EOL`?

Reply:
for now since this is only available on windows and I don't know if dolphin outputs `CRLF` always or `LF` on unix, we will leave it as is